api/seedreconfig/seedreconfig.go

```diff
@@ -0,0 +1,86 @@
+package seedreconfig
+
+type PEM string
+
+const (
+	SeedReconfigVersion = 1
+)
+
+// SeedReconfiguration contains all the information that is required to
+// transform a machine started from a SNO seed image (which contains dummy seed
+// configuration) into a SNO cluster with a desired configuration. During an
+// IBU, this information is taken from the cluster that is being upgraded (the
+// original SNO's LCA writes a file to the seed stateroot to communicate this
+// information). During an IBI, this information is placed in the configuration
+// ISO created by the cluster-relocation-operator.
+//
+// WARNING: Changes to this struct and its sub-structs should not be made
+// lightly, as it is also used by the cluster-relocation-operator. Any changes
+// made here will also need to be handled in the cluster-relocation-operator or
+// be backwards compatible. If you've made a breaking change, you will need to
+// increment the SeedReconfigVersion constant to avoid silently breakage and
+// allow for backwards compatibility code.
+type SeedReconfiguration struct {
+	// The version of the SeedReconfiguration struct format. This is used to detect
+	// breaking changes to the struct.
+	APIVersion int `json:"api_version"`
+
+	// The desired base domain for the cluster. Equivalent to install-config.yaml's baseDomain.
+	// This will replace the base domain of the seed cluster.
+	BaseDomain string `json:"base_domain,omitempty"`
+
+	// The desired cluster name for the cluster. Equivalent to install-config.yaml's clusterName.
+	// This will replace the cluster name of the seed cluster.
+	ClusterName string `json:"cluster_name,omitempty"`
+
+	// The desired cluster-ID. During an IBU, this is the ID of the original
+	// SNO. During an IBI, this can either be empty, in which case a new
+	// cluster-ID will be generated by LCA, or it can be set to the ID of the
+	// new cluster, in case one had to be pre-generated for some reason.
+	ClusterID string `json:"cluster_id,omitempty"`
+
+	// The desired IP address of the SNO node.
+	NodeIP string `json:"node_ip,omitempty"`
+
+	// ?
+	ReleaseRegistry string `json:"release_registry,omitempty"`
+
+	// The desired hostname of the SNO node.
+	Hostname string `json:"hostname,omitempty"`
+
+	// Whether or not the cluster is using a mirror registry.
+	MirrorRegistry bool `json:"mirror_registry,omitempty"`
```

Comment:
Any idea why we need to know this?

Reply:
In order to understand if cluster uses mirror registry or not as this is sort of tricky. This field is helping to get decision if we need to override seed release registry or not

Reply:
If you get ICSP/IDMS you are using a mirror registry.
Why do we need this extra field?
Also, it seems that in any case if the registry in the cluster version deployment is different than the one in the SeedReconfiguration (and it's not resolvable via IDMS/ICSP) we will override the clusterversion deployment with the new registry.
https://github.com/openshift-kni/lifecycle-agent/blob/7c5804dfbe03d7af0081d5b8b6976895f13a48d5/ibu-imager/postpivot/postpivot.go#L287

Reply:
Seems that this isn't really used
https://github.com/search?q=repo%3Aopenshift-kni%2Flifecycle-agent%20MirrorRegistry&type=code

Reply:
`MirrorRegistry` is the old name, it was part of this PR only because I started this PR before the name change. I've since rebased